### PR TITLE
Add timeout for ccr recovery action

### PR DESCRIPTION
**Pull request by Tim-Brooks**
This is related to #35975. It adds a action timeout setting that allows
timeouts to be applied to the individual transport actions that are
used during a ccr recovery.

### CODE REVIEW

**Comment by tbrooks8** [comment]
@ywelsch 

1. What are you thoughts on the default value for this?
2. I'm not sure how we want to pursue testing? It looks like we do not test the individual action timeouts for normal recovery. I could insert a wait or something using the `CcrRestoreSourceService` callbacks. But I had planned on removing those eventually as it seems weird to have callbacks dedicated for testing purposes.

**Review by ywelsch — COMMENTED** [review]
Code and default value looks good (we can always increase it later if we see the need for it). For testing, you could set a low timeout, and then inject a failure at an arbitrary point (i.e. randomly on one of the CCR recovery actions) using `MockTransportService`, see `RecoveryActionBlocker` as an example in `IndexRecoveryIT`, and then check that the restore fails in a timely fashion.

File: x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/CcrRepositoryIT.java [hunk 1]
```diff
@@ -308,6 +313,78 @@ public void testRateLimitingIsEmployed() throws Exception {
...
+
+
+        for (MockTransportService transportService : transportServices) {
+            transportService.clearAllRules();
+        }
+
+        settingsRequest = new ClusterUpdateSettingsRequest();
+        TimeValue defaultValue = CcrSettings.INDICES_RECOVERY_ACTION_TIMEOUT_SETTING.getDefault(Settings.EMPTY);
+        settingsRequest.persistentSettings(Settings.builder().put(CcrSettings.INDICES_RECOVERY_ACTION_TIMEOUT_SETTING.getKey(),
+            defaultValue));
```
Comment (ywelsch) [inline x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/CcrRepositoryIT.java]: setting it to null also resets to the default

File: x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/CcrRepositoryIT.java [hunk 2]
```diff
@@ -308,6 +313,78 @@ public void testRateLimitingIsEmployed() throws Exception {
         }
     }
 
+    public void testIndividualActionsTimeout() throws Exception {
+        ClusterUpdateSettingsRequest settingsRequest = new ClusterUpdateSettingsRequest();
+        TimeValue timeValue = new TimeValue(100);
```
Comment (ywelsch) [inline x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/CcrRepositoryIT.java]: I prefer `TimeValue.timeValueMillis(100)`. It's clearer what the 100 are.

**Review by ywelsch — APPROVED** [review]
LGTM

**Pull request merged**